Source/WebCore/workers/service/ServiceWorkerRegistration.cpp

```diff
@@ -352,8 +353,67 @@ void ServiceWorkerRegistration::getNotifications(const GetNotificationOptions& f
 CookieStoreManager& ServiceWorkerRegistration::cookies()
 {
     if (!m_cookieStoreManager)
-        m_cookieStoreManager = CookieStoreManager::create();
+        m_cookieStoreManager = CookieStoreManager::create(*this);
     return *m_cookieStoreManager;
 }
 
+void ServiceWorkerRegistration::addCookieChangeSubscriptions(Vector<CookieStoreGetOptions>&& subscriptions, Ref<DeferredPromise>&& promise)
+{
+    Vector<CookieChangeSubscription> cookieChangeSubscriptions;
+    for (auto& subscription : subscriptions) {
+        if (isContextStopped()) {
+            promise->reject(Exception { ExceptionCode::InvalidStateError, "The script execution context is not currently running"_s });
+            return;
+        }
+
+        // FIXME: Fall back to scope url since spec does not specify an alternative and WPT layout tests expect this behavior (https://github.com/WICG/cookie-store/issues/236).
+        String url;
+        if (subscription.url.isNull())
+            url = scope();
+        else
+            url = URL(scriptExecutionContext()->url(), subscription.url).string();
```

Comment:
I tried using `completeURL()` originally but encountered an issue. The WPT test `/LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscribe_arguments.https.any.js` has a test that calls `CookieStore::Subscribe` without a url and expects it to pass. Since this test is passing in a null url, passing it to `completeURL()` resulted in a null result for the url and as per the spec, I returned a TypeError. Since the spec and test seemed to be conflicting, I uploaded this patch as a draft and asked about this. Sihui responded here (https://github.com/WebKit/WebKit/pull/31979#issuecomment-2279485767). She also mentioned that we should file a spec issue about this and use a FIXME for now. Which is what I did and is why we need to use `URL()` here. Here is the spec issue we filed: https://github.com/WICG/cookie-store/issues/236

Reply:
You're dealing with the null URL above:
```
if (subscription.url.isNull())
            url = scope();
```
So I don't understand your answer.

Reply:
Oh right--thank you for pointing that out. Fixed!